metricproviders/plugin/plugin.go

```diff
@@ -0,0 +1,59 @@
+package plugin
+
+import (
+	"github.com/argoproj/argo-rollouts/metric"
+	"github.com/argoproj/argo-rollouts/metricproviders/plugin/client"
+	"github.com/argoproj/argo-rollouts/metricproviders/plugin/rpc"
+	"github.com/argoproj/argo-rollouts/pkg/apis/rollouts/v1alpha1"
+)
+
+const ProviderType = "RPCPlugin"
+
+type MetricPlugin struct {
```

Comment:
Not user I understand the reason why this struct exists but it is currently mainly most of the operations to the `rpc.MetricsPlugin`. Why not just declare the struct like below and avoid having to implement all wrapper methods?
```golang
type MetricPlugin struct {
	rpc.MetricsPlugin
	metric.Provider
}
```

Reply:
Yea I changed this to just depend on the forwarded functions got to deleted some code :D